### ISSUE TITLE
Support for Structures

### ISSUE THREAD

**Issue by lstagner** [BobPortmann/IDLCall.jl#4]
Now that the install process is pretty much automatic one of the last major shortcomings is the lack of support for IDL structures.

Once IDLCall can support structures I think it will be ready for the IDL community.

**Comment by BobPortmann** [comment]
Yes, the structures issue is important but tricky. That is part of the reason I punted on it. The problem is, what should they map to?

`types` or `immutable`: in some sense this is the closest analog in julia (IDL structures are essentially C structs under the hood and `immutable` is probably closest to that) but `types` are more heavyweight in the sense that they are difficult to add on to. In IDL, I use `create_struct` all the time to add on to structures but on the julia side this would mean making a new `type`. I don't think this is workable.

`Dict` or `OrderedDict`: this could work but then what would a IDL hash, orderedhash, or dictionary map to? Maybe this is ok, you would just have to say which julia type to map the IDL type to when you `put_var` (and vice-versa). But you would have to copy the data, which is necessary anyway in the `RCP` case anyway.

Custom `type` based on `Associative`: this could be as simple as a loose wrapper around the OrderedDict code (or copying that code and renaming the type) or a fancier type that used the IDL C structs directly under the hood. In that case the memory could be shared in the Callable case but a copy would still be necessary in the RCP case.

Thoughts?

ps I don't necessarily think we need to wait on this issue to announce IDLCall. It is pretty handy as is. Also, I am going to file several issue that need working on but none of them really have to hold anything up.

**Comment by lstagner** [comment]
> Custom type based on Associative: this could be as simple as a loose wrapper around the OrderedDict code (or copying that code and renaming the type) or a fancier type that used the IDL C structs directly under the hood. In that case the memory could be shared in the Callable case but a copy would still be necessary in the RCP case.

I think this is the best solution. We should also do it for the other IDL types that do not have an obvious Julia equivalent.
